src/transformers/generation/flax_logits_process.py

```diff
@@ -259,10 +259,90 @@ def __init__(self, min_length: int, eos_token_id: int):
         apply_penalty = 1 - jnp.clip(cur_len - self.min_length, 0, 1)
 
         scores = jnp.where(apply_penalty, scores.at[:, self.eos_token_id].set(-float("inf")), scores)
 
         return scores
+
+
+class FlaxSuppressTokensAtBeginLogitsProcessor:
+    r"""
+    Args:
+    [`FlaxLogitsProcessor`] supressing a list of tokens at an index.
+        begin_suppress_tokens (`list`):
+            Tokens to not sample.
+        begin_index (`int`):
+            Index where the tokens are suppressed.
+    """
+
+    def __init__(self, begin_suppress_tokens, begin_index):
+        self.begin_suppress_tokens = list(begin_suppress_tokens)
+        self.begin_index = begin_index
+
+    def __call__(self, input_ids, scores, cur_len: int):
+        apply_penalty = 1 - jnp.bool_(cur_len - self.begin_index)
+
+        scores = jnp.where(apply_penalty, scores.at[:, self.begin_suppress_tokens].set(-float("inf")), scores)
+
+        return scores
+
+
+class FlaxSuppressTokensLogitsProcessor(FlaxLogitsProcessor):
+    r"""
+    Args:
+    [`FlaxLogitsProcessor`] supressing a list of tokens.
+        suppress_tokens (`list`):
+            Tokens to not sample.
+    """
+
+    def __init__(self, suppress_tokens: list):
+        self.suppress_tokens = list(suppress_tokens)
+
+    def __call__(self, input_ids: jnp.ndarray, scores: jnp.ndarray, cur_len: int) -> jnp.ndarray:
+        scores = scores.at[..., self.suppress_tokens].set(-float("inf"))
+
+        return scores
+
+
+class FlaxForceTokenAtIdxLogitsProcessor(FlaxLogitsProcessor):
+    r"""
+    Args:
+    [`FlaxLogitsProcessor`] that forces a token to be sampled at an index.
+        apply_idx (`int`):
+            Index where sampling is forced.
+        token_id (`int`):
+            Token that is forced to be sampled.
+    """
+
+    def __init__(self, apply_idx: int, token_id: int):
+        self.apply_idx = apply_idx
+        self.token_id = token_id
+
+    def __call__(self, input_ids: jnp.ndarray, scores: jnp.ndarray, cur_len: int) -> jnp.ndarray:
+        new_scores = jnp.full(scores.shape, -float("inf"))
+
+        apply_penalty = 1 - jnp.bool_(cur_len - self.apply_idx)
+
+        scores = jnp.where(apply_penalty, new_scores.at[:, self.token_id].set(0), scores)
+
+        return scores
+
+
+class FlaxForceTokensLogitsProcessor(FlaxLogitsProcessor):
+    r"""
+    Args:
+    [`FlaxLogitsProcessor`] that forces tokens to be sampled at given indices.
+        force_token_map (`list`):
+            Map giving token ids and indices where they will be forced to be sampled.
+    """
+
+    def __init__(self, force_token_map):
+        self.processors = [FlaxForceTokenAtIdxLogitsProcessor(i[0], i[1]) for i in force_token_map]
+
+    def __call__(self, input_ids: jnp.ndarray, scores: jnp.ndarray, cur_len: int) -> jnp.ndarray:
+        for processor in self.processors:
+            scores = processor(input_ids, scores, cur_len)
+
+        return scores
```

Comment:
```suggestion
    def __call__(self, input_ids: jnp.ndarray, scores: jnp.ndarray, cur_len: int) -> jnp.ndarray:
        current_token = self.force_token_map.get(cur_len, None)
         if current_token is not None:
             scores = jnp.full(scores.shape, -float("inf"))
             scores.at[:, current_token].set(0)
        return scores
```

Reply:
This folllows the pytorch implementation and is much easier to understand

Reply:
This won't work under jax.jit/pmap/pjit because cur_len will be a tracer

Reply:
WDYT @ArthurZucker @sanchit-gandhi ?

Reply:
@ArthurZucker just bumping this conversation

Reply:
A possible alternative would be to craft this function [as we did in TF](https://github.com/huggingface/transformers/blob/588faad1062198e45cf3aebed21dc1fc1e1ed0d7/src/transformers/generation/tf_logits_process.py#L549), based on the [`cond`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.cond.html) function (which behaves the same way in TF and JAX).

Having the same implementation goes a long way for our text generation maintainers 🙏